io/src/main/scala/fs2/io/net/SocketGroup.scala

```diff
@@ -177,8 +179,17 @@ private[net] object SocketGroup {
             }
 
           Stream.eval(acceptChannel.attempt).flatMap {
```

Comment:
Note that the leak of a single socket when being interrupted is still present here. I tried to replace this `eval` with a `Stream.resource` backed by an underlying `Resource.makeFull` which polled over the `sch.accept` operation, but it didn't seem like it responded to interruption immediately.

Reply:
Hm, that should work fine...

Reply:
I'll update with a snippet sometime today

Reply:
Looks like this was because of what Fabio described in his comment (wasn't polling on the `allocated` resource)